packages/core/src/parameters/getParameterDefinitionsFromSource.js

```diff
@@ -0,0 +1,174 @@
+        i++
+        continue
+      }
+    }
+
+    prevIndent = lines[i].indent
+    if (isGroup) {
+      // group
+      const name = '_group_' + (groupIndex++)
+      const def = parseComment(code, lineNum, name)
+      let caption = def.caption
+      if (caption[0] === '>') {
+        caption = caption.substring(1).trim()
+        if (!def.options) def.options = {}
+        def.options.initial = 'closed'
+      }
+
+      defs.push(prev = { name, type: 'group', caption, ...def.options })
+    } else {
+      const idx = code.indexOf('/')
+      if (idx === -1) {
+        const def = parseDef(code, lineNum)
+        def.caption = def.name
+        defs.push(prev = def)
+      } else {
+        defs.push(prev = parseOne(
+          code.substring(idx).trim(),
+          code.substring(0, idx).trim(),
+          lineNum, lineNum
+        ))
+      }
+    }
+    i++
+  }
+
+  return defs
+}
+
+const parseOne = (comment, code, line1, line2) => {
+  let def = parseDef(code, line2)
+  const { caption, options } = parseComment(comment, line1, def.name)
+  def.caption = caption || def.name
+  if (options) {
+    def = { ...def, ...options }
+    if (def.type === 'checkbox' && 'initial' in def) def.checked = true
+    if (def.type === 'slider') {
+      if (def.min === undefined) {
+        def.min = 0
+      }
+      if (def.max === undefined) {
+        def.max = 100
+      }
+    }
+  }
+
+  return def
+}
+
+const extractTextFromComment = (c) => {
+  const prefix = c.substring(0, 2)
+  if (prefix === '//') c = c.substring(2)
+  if (prefix === '/*') c = c.substring(2, c.length - 2)
+
+  return c.trim()
+}
+
+const parseComment = (comment, line, paramName) => {
+  comment = extractTextFromComment(comment)
+
+  const ret = {}
+  const idx = comment.indexOf('{')
+  if (idx !== -1) {
+    try {
+      ret.options = JSON5.parse(comment.substring(idx))
+    } catch (e) {
+      throw new EvalError(`${e.message}, parameter:${paramName}, line:${line}: ${comment.substring(idx)}`, 'code', line)
+    }
+    comment = comment.substring(0, idx).trim()
+  }
+
+  ret.caption = comment
+
+  return ret
+}
+
+const parseDef = (code, line) => {
+  if (code[code.length - 1] === ',') code = code.substring(0, code.length - 1).trim()
+  let idx = code.indexOf('=')
+
+  if (idx === -1) idx = code.indexOf(':')
+
+  if (idx === -1) {
+    return { name: code, type: 'text' }
+  } else {
+    const initial = code.substring(idx + 1).trim()
+
+    const ret = { type: 'text', name: code.substring(0, idx).trim() }
+
+    if (initial === 'true' || initial === 'false') {
+      ret.type = 'checkbox'
+      ret.checked = initial === 'true'
+    } else if (/^[0-9]+$/.test(initial)) {
+      ret.type = 'int'
+      ret.initial = parseFloat(initial)
+    } else if (/^[0-9]+\.[0-9]+$/.test(initial)) {
+      ret.type = 'number'
+      ret.initial = parseFloat(initial)
+    } else {
+      try {
+        ret.initial = JSON5.parse(initial)
```

Comment:
Hmm... I doubt JSON5 will every be adopted, but this does simply the code. And this is only for the WEB based UI as well. So, OK.

Reply:
In the contrary, my perception is that it is very well adopted https://www.npmjs.com/package/json5 37,466,988 weekly downloads. I found about it while writing babel plugins, babel uses it internally, and many libraries that want to for example use json for config files but want cleaner syntax that is closer to JS code. :)